src/components/FileViewer/FileList.tsx

```diff
@@ -42,11 +56,119 @@ export default function FileList({
     );
   };
 
+  const boundingDivRef = useRef<HTMLDivElement>(null) 
+  const contextMenuRef = useRef<HTMLDivElement>(null) 
+  const [mousePos, setMousePos] = useState<{x: number, y: number}>({x: 0, y: 0});
+  const [showContextMenu, setShowContextMenu] = useState(false);
+  const [contextMenuCoords, setContextMenuCoords] = useState({x: 0, y: 0});
+  const [contextMenuFile, setContextMenuFile] = useState<ClientFile>();
+  const [absCoords, setAbsCoords] = useState({x: 0, y: 0})
+  const [boundingDivDimensions, setBoundingDivDimensions] = useState({ height: 0, width: 0})
+
+  const contextMenuDimensions = { height: 234, width: 176 } 
```

Comment:
I see, that makes sense.
I'm not too sure if `useLayoutEffect` will exactly work though, since that fires before painting, AKA the context menu hasn't rendered yet.

But also looks like this number is currently only used for preventing the context menu from going out of bounds, so it's not too high stake. Leaving it hard coded might be fine then.